### DISCUSSION TITLE
ConnectionTimeout is set. It doesn't seem to work

### DISCUSSION THREAD

**Discussion by jahnli** [PhilipsHue/flutter_reactive_ble#381] [Q&A]
![image](https://user-images.githubusercontent.com/20943608/134809703-8bc09dee-66eb-427c-be02-17df28076ce1.png)

When I set up a second overtime, it will not be disconnected or return an error, bluetooth status has also been a DeviceConnectionState. Connecting

How do I catch mistakes, how to terminate DeviceConnectionState. Connecting

**Comment by atauf** [comment]
I'm not exactly positive I'm following, but in order to terminate 'DeviceConnectionState.connection you can just call .disconnect(deviceId). You'll need to use

```
Provider.of<BleDeviceConnector>(context, listen: false)
              .disconnect(deviceId);
```
